### PR TITLE
Adding dangling line's generation part mapping 

### LINKED ISSUE #793: Optional dangling line generation part
### Describe the current behavior

A dangling line can have a generation part. It seems that these information are not exposed in pypowsybl. 

### Describe the expected behavior

Should be exposed.

### Describe the motivation

_No response_

### Extra Information

_No response_

### PR DESCRIPTION
**Pull request by HugoKulesza**
update when no generation created to do

**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

Fixing #793 

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by HugoKulesza** [comment]
It is currently not possible to add a new Generation in a DanglingLine already created without one (the GenerationAdder can only be called inside a DanglingLineAdder when creating a DanglingLine)
Should we :
- add a method in powsybl-core to be able to add a Generation on a already created dangling line ?
- or in pypowsybl recreate entirely the dangling line when we try to add a generation part on it ?
@geofjamg

File: java/src/main/java/com/powsybl/dataframe/network/NetworkDataframes.java [hunk 1]
```diff
@@ -28,6 +28,7 @@
 
 import static com.powsybl.dataframe.MappingUtils.*;
 import static com.powsybl.dataframe.network.PerUnitUtil.*;
+import static java.lang.Double.NaN;
```
Comment (geofjamg) [inline java/src/main/java/com/powsybl/dataframe/network/NetworkDataframes.java]: Could we avoid this useless change ?

File: pypowsybl/network/impl/network.py [hunk 2]
```diff
@@ -1244,14 +1244,20 @@ def get_dangling_lines(self, all_attributes: bool = False, attributes: List[str]
               - **ucte-xnode-code**: deprecated for pairing key.
               - **paired**: if the dangling line is paired with a tie line
               - **tie_line_id**: the ID of the tie line if the dangling line is paired
+              - **min_p** (optional): Minimum active power output of the dangling line's generation part
```
Comment (geofjamg) [inline pypowsybl/network/impl/network.py]: To revert ?

File: pypowsybl/network/impl/network.py [hunk 3]
```diff
@@ -1293,6 +1299,35 @@ def get_dangling_lines(self, all_attributes: bool = False, attributes: List[str]
         """
         return self.get_elements(ElementType.DANGLING_LINE, all_attributes, attributes, **kwargs)
 
+    def get_dangling_lines_generation(self, all_attributes: bool = False, attributes: List[str] = None,
```
Comment (geofjamg) [inline pypowsybl/network/impl/network.py:1366]: what is the purpose  of **kwargs ?

File: pypowsybl/network/impl/network.py [hunk 4]
```diff
@@ -3939,17 +4007,18 @@ def create_batteries(self, df: DataFrame = None, **kwargs: ArrayLike) -> None:
         """
         return self._create_elements(ElementType.BATTERY, [df], **kwargs)
 
-    def create_dangling_lines(self, df: DataFrame = None, **kwargs: ArrayLike) -> None:
+    def create_dangling_lines(self, df: DataFrame = None, generation_df: DataFrame = pd.DataFrame(), **kwargs: ArrayLike) -> None:
```
Comment (geofjamg) [inline pypowsybl/network/impl/network.py:4073]: Is **kwargs stil working for example to pass the dangling base attributes inline and the generation part with generation_df ? 
```python
n.create_dangling_line(id='X', r=3.4, ..., generation_df=...)
```

Comment (HugoKulesza) [reply pypowsybl/network/impl/network.py:4073]: Yes, the tests use create_dangling_lines this way to verify it still works.
On the other end, data for generation_df cannot be passed using kwargs

**Review by geofjamg — APPROVED** [review]

**Pull request merged**